pkg/3scale/amp/operator/upgrade.go

```diff
@@ -50,9 +50,43 @@ func (u *UpgradeApiManager) Upgrade() (reconcile.Result, error) {
 		}
 	}
 
+	// TODO should there be some kind of coordination/wait between the ImageStream
+	// tags creation and the trigger update/swap ??
```

Comment:
See this comment

Reply:
Good question. I would expect that OCP takes care of all going behind the scenes and, thus, we do not need to wait. 

If imagestream new tags are not available when we change image trigger, I expect the DC would fail to "evaluate" the new image and will retry eventually.

Thus, I would not implement wait, and keep an eye open just in case we find issues.

Reply:
:+1: